### PR TITLE
Better error messages when coupons are stuck in a pending order.

### LINKED ISSUE #27673: Coupon "usage limit has been reached" after a failed payment
**Prerequisites (mark completed items with an [x]):**
- [X] I have have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**
Coupons with a usage limit are showing "usage limit has been reached" after a failed payment using that coupon.

**Expected behavior**
Coupon should still be usable after a failed payment. 

**Actual behavior**
Applying a coupon to an order (specifically with a usage limit of 1) and making a failed payment at checkout places a `_maybe_used_by` lock on the coupon itself - making it unusable by the same person.

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Steps to reproduce the bug:
1. Create a coupon with a usage limit of 1
2. Add it to your cart
3. Checkout in a way that causes a failed payment
4. Coupon now gets removed as 'invalid'
5. Try re-add the coupon, you'll be told "usage limit has been reached"

**Screenshots**
image
image
image
image
image

**Isolating the problem (mark completed items with an [x]):**
- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://www.f*******e.com
Site address (URL): https://www.f*******e.com
WC Version: 4.5.1
REST API Version: ✔ 4.5.1
WC Blocks Version: ✔ 3.1.0
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.5.0
Log Directory Writable: ❌ To allow logging
make /var/www/html/src/wp-content/uploads/wc-logs/ writable or define a custom WC_LOG_DIR.

WP Version: 5.5.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx/1.18.0
PHP Version: 7.4.9
PHP Post Max Size: 10 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.61.1
OpenSSL/1.0.2k

SUHOSIN Installed: –
MySQL Version: 5.6.10
Max Upload Size: 10 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.5.1
WC Database Prefix: wp_
Total Database Size: 5346.26MB
Database Data Size: 3706.20MB
Database Index Size: 1640.06MB
wp_woocommerce_sessions: Data: 14.03MB + Index: 1.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_order_items: Data: 25.55MB + Index: 8.52MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 448.98MB + Index: 387.97MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.08MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.11MB + Index: 0.14MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.11MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 1.52MB + Index: 0.69MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
fakenames: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
w****o_products: Data: 13.52MB + Index: 2.84MB + Engine InnoDB
wp_actionscheduler_actions: Data: 10.03MB + Index: 17.13MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 8.02MB + Index: 8.03MB + Engine InnoDB
wp_as3cf_items: Data: 13.52MB + Index: 23.09MB + Engine InnoDB
wp_blc_filters: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_blc_instances: Data: 0.05MB + Index: 0.05MB + Engine InnoDB
wp_blc_links: Data: 0.22MB + Index: 0.06MB + Engine InnoDB
wp_blc_synch: Data: 0.27MB + Index: 0.14MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 115.66MB + Index: 81.05MB + Engine InnoDB
wp_gf_draft_submissions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_gf_entry: Data: 4.52MB + Index: 0.58MB + Engine InnoDB
wp_gf_entry_meta: Data: 11.52MB + Index: 17.56MB + Engine InnoDB
wp_gf_entry_notes: Data: 3.52MB + Index: 0.56MB + Engine InnoDB
wp_gf_form: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_gf_form_meta: Data: 0.30MB + Index: 0.00MB + Engine InnoDB
wp_gf_form_revisions: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_gf_form_view: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mclean_refs: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
wp_mclean_scan: Data: 0.03MB + Index: 0.03MB + Engine InnoDB
wp_oauth_access_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_oauth_authorization_codes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_oauth_jwt: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_oauth_public_keys: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_oauth_refresh_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_oauth_scopes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 29.22MB + Index: 10.08MB + Engine InnoDB
wp_oses_attachments: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_oses_clicks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_oses_emails: Data: 1666.17MB + Index: 6.48MB + Engine InnoDB
wp_oses_email_attachments: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_oses_failures: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
wp_oses_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_pmxi_files: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_pmxi_history: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_pmxi_imports: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_pmxi_posts: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_pmxi_templates: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_postmeta: Data: 952.66MB + Index: 842.36MB + Engine InnoDB
wp_posts: Data: 105.23MB + Index: 43.45MB + Engine InnoDB
wp_redirection_404: Data: 5.03MB + Index: 14.13MB + Engine InnoDB
wp_redirection_groups: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_redirection_items: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_redirection_logs: Data: 0.23MB + Index: 0.17MB + Engine InnoDB
wp_termmeta: Data: 1.52MB + Index: 0.78MB + Engine InnoDB
wp_terms: Data: 0.23MB + Index: 0.30MB + Engine InnoDB
wp_terms_back: Data: 0.13MB + Index: 0.13MB + Engine InnoDB
wp_term_relationships: Data: 13.55MB + Index: 8.47MB + Engine InnoDB
wp_term_relationships_back: Data: 7.52MB + Index: 4.39MB + Engine InnoDB
wp_term_taxonomy: Data: 0.22MB + Index: 0.27MB + Engine InnoDB
wp_term_taxonomy_back: Data: 0.06MB + Index: 0.09MB + Engine InnoDB
wp_usermeta: Data: 130.45MB + Index: 81.17MB + Engine InnoDB
wp_users: Data: 7.41MB + Index: 7.25MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.22MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 3.52MB + Index: 3.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 10.52MB + Index: 12.00MB + Engine InnoDB
wp_wc_order_stats: Data: 5.52MB + Index: 4.53MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 2.52MB + Index: 3.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 7.52MB + Index: 13.83MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_warranty_products: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta_back: Data: 0.06MB + Index: 0.08MB + Engine InnoDB
wp_woocommerce_pickup_locations_geodata: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_yith_wcwl: Data: 2.06MB + Index: 0.09MB + Engine InnoDB
wp_yith_wcwl_lists: Data: 2.17MB + Index: 4.31MB + Engine InnoDB
wp_yoast_indexable: Data: 54.61MB + Index: 26.61MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Data: 2.39MB + Index: 2.36MB + Engine InnoDB
wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.38MB + Index: 0.22MB + Engine InnoDB
wp_yoast_seo_meta: Data: 13.36MB + Index: 0.00MB + Engine InnoDB
ziplatlng: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
_mig_wpmdb_alter_statements: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

acf-field: 26
acf-field-group: 5
attachment: 65944
blocks: 17
custom_css: 1
li_event: 366
nav_menu_item: 369
oembed_cache: 17
page: 64
post: 331
product: 21325
product_variation: 13904
revision: 623
shop_coupon: 2927
shop_order: 94529
shop_order_refund: 11094
store: 21
wc_order_status: 7
wc_pickup_location: 18
wll_records: 589
wo_client: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (33) ###

Gravity Forms: by Gravity Forms – 2.4.20
Admin Menu Editor Pro: by Janis Elsts – 2.12.2
Advanced Custom Fields PRO: by Elliot Condon – 5.9.0
WP Offload Media: by Delicious Brains – 2.4.2
Application Passwords: by George Stephanis – 0.1.2
Classic Editor: by WordPress Contributors – 1.6
Custom Post Type UI: by WebDevStudios – 1.8.0
Geolocation IP Detection: by Yellow Tree (Benjamin Pick) – 3.1.1
Go Live Update Urls: by OnPoint Plugins – 6.1.2
Lucky Orange | Chat, heatmaps, polls, visitor recordings, live analytics.: by LuckyOrange.com – 1.98
Members: by MemberPress – 3.0.10
PixelYourSite PRO: by PixelYourSite – 7.6.5 – Installed version not tested with active version of WooCommerce 4.5.1
Product Catalog Feed Pro by PixelYourSite: by PixelYourSite – 4.1.3 – Installed version not tested with active version of WooCommerce 4.5.1
PWD Content Sync Receiver: by Portland Web Design – 1.0.6
Redirection: by John Godley – 4.8
Redis Cache Pro: by Till Krüss – 1.9.0
Regenerate Thumbnails Advanced: by ShortPixel – 2.3.2
Safe SVG: by Daryll Doyle – 1.9.9
User Switching: by John Blackbourn & contributors – 1.5.5
When Last Login: by Yoohoo Plugins – 1.2
WooCommerce AvaTax [w****o Modified - DO NOT UPDATE AUTOMATICALLY]: by SkyVerge – 1.10.2 (update to version 1.10.3 is available) – Installed version not tested with active version of WooCommerce 4.5.1
WooCommerce Brands: by WooCommerce – 1.6.19 – Installed version not tested with active version of WooCommerce 4.5.1
WooCommerce Authorize.Net Gateway: by SkyVerge – 3.3.0 – Installed version not tested with active version of WooCommerce 4.5.1
WooCommerce PayPal Checkout Gateway: by WooCommerce – 2.0.3 – Installed version not tested with active version of WooCommerce 4.5.1
WooCommerce UPS Shipping: by WooCommerce – 3.2.28 – Installed version not tested with active version of WooCommerce 4.5.1
WooCommerce Smart Coupons: by StoreApps – 4.8.0 (update to version 4.8.1 is available) – Installed version not tested with active version of WooCommerce 4.5.1
WooCommerce: by Automattic – 4.5.1 (update to version 4.5.2 is available)
WooSidebars: by WooCommerce – 1.4.5
Yoast SEO: by Team Yoast – 14.8.1 – Installed version not tested with active version of WooCommerce 4.5.1
WP Crontrol: by John Blackbourn & crontributors – 1.8.5
WP Migrate DB Pro: by Delicious Brains – 1.9.10
WP Offload SES: by Delicious Brains – 1.5.3
YITH WooCommerce Anti-Fraud Premium: by YITH – 1.1.9 – Installed version not tested with active version of WooCommerce 4.5.1

### Inactive Plugins (10) ###

Bugsnag Error Monitoring: by Bugsnag Inc. – 1.5.0
GDPR Cookie Consent: by WebToffee – 1.9.0
PixelYourSite Pinterest: by PixelYourSite – 2.0.7 – Installed version not tested with active version of WooCommerce 4.5.1
PWD Flipping Book Scraper: by Portland Web Design – 1.0.1
PWD Gravity Forms Monday.com Integration: by Portland Web Design – 1.1
PW WooCommerce Gift Cards Pro: by Pimwick
LLC – 1.245 – Installed version not tested with active version of WooCommerce 4.5.1

Query Monitor: by John Blackbourn – 3.6.4
ShortPixel Image Optimizer: by ShortPixel – 4.20.2
WP Rocket [Jareds Version 07/12/19]: by WP Media – 3.4.2.2
YITH WooCommerce Wishlist: by YITH – 3.0.13 – Installed version not tested with active version of WooCommerce 4.5.1

### Dropin Plugins (1) ###

object-cache.php: Redis Cache Pro

### Must Use Plugins (1) ###

Purge Cache: by The Code Company – 1.0

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #74 - /shop
Cart: #75 - /cart
Checkout: #76 - /checkout
My account: #77 - /my-account
Terms and conditions: ❌ Page not set

### Theme ###

Name: w****o
Version: 3.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.6.2
Parent Theme Author URL: http://www.uxthemes.com/
WooCommerce Support: ✔

### Templates ###

Overrides: w****o-child/woocommerce/archive-product.php
w****o-child/woocommerce/cart/cart-empty.php
w****o-child/woocommerce/cart/cart-shipping.php
w****o-child/woocommerce/cart/cart-totals.php
w****o-child/woocommerce/cart/cart.php
w****o-child/woocommerce/cart/mini-cart.php
w****o-child/woocommerce/cart/proceed-to-checkout-button.php
w****o-child/woocommerce/checkout/form-billing.php
w****o-child/woocommerce/checkout/form-checkout.php
w****o-child/woocommerce/checkout/form-coupon.php
w****o-child/woocommerce/checkout/form-login.php
w****o-child/woocommerce/checkout/form-shipping.php
w****o-child/woocommerce/checkout/review-order.php
w****o-child/woocommerce/checkout/thankyou.php
w****o-child/woocommerce/content-product.php
w****o-child/woocommerce/content-product_cat.php
w****o-child/woocommerce/content-single-product.php
w****o-child/woocommerce/content-widget-product.php
w****o-child/woocommerce/emails/customer-new-account.php
w****o-child/woocommerce/emails/customer-on-hold-order.php
w****o-child/woocommerce/emails/customer-processing-order.php
w****o-child/woocommerce/emails/customer-reset-password.php
w****o-child/woocommerce/emails/email-footer.php version 2.3.0 is out of date. The core version is 3.7.0
w****o-child/woocommerce/emails/email-header.php version 2.4.0 is out of date. The core version is 4.0.0
w****o-child/woocommerce/emails/email-order-details.php
w****o-child/woocommerce/emails/email-order-items.php
w****o-child/woocommerce/emails/email-styles.php version 3.3.0 is out of date. The core version is 4.0.0
w****o-child/woocommerce/global/breadcrumb.php
w****o-child/woocommerce/global/quantity-input.php
w****o-child/woocommerce/global/wrapper-end.php
w****o-child/woocommerce/global/wrapper-start.php
w****o-child/woocommerce/loop/loop-end.php
w****o-child/woocommerce/loop/loop-start.php
w****o-child/woocommerce/loop/orderby.php
w****o-child/woocommerce/loop/pagination.php
w****o-child/woocommerce/loop/result-count.php
w****o-child/woocommerce/loop/sale-flash.php
w****o-child/woocommerce/myaccount/form-edit-account.php
w****o-child/woocommerce/myaccount/form-edit-address.php
w****o-child/woocommerce/myaccount/form-login.php
w****o-child/woocommerce/myaccount/form-lost-password.php
w****o-child/woocommerce/myaccount/my-address.php
w****o-child/woocommerce/myaccount/navigation.php
w****o-child/woocommerce/myaccount/orders.php
w****o-child/woocommerce/myaccount/view-order.php
w****o-child/woocommerce/notices/error.php
w****o-child/woocommerce/notices/notice.php
w****o-child/woocommerce/notices/success.php
w****o-child/woocommerce/order/order-details-item.php
w****o-child/woocommerce/order/order-details.php version 3.5.2 is out of date. The core version is 3.7.0
w****o-child/woocommerce/product-searchform.php
w****o-child/woocommerce/single-product/add-to-cart/simple.php
w****o-child/woocommerce/single-product/add-to-cart/variable.php
w****o-child/woocommerce/single-product/add-to-cart/variation-add-to-cart-button.php
w****o-child/woocommerce/single-product/add-to-cart/variation.php
w****o-child/woocommerce/single-product/meta.php
flatsome/woocommerce/single-product/photoswipe.php
w****o-child/woocommerce/single-product/price.php
w****o-child/woocommerce/single-product/product-image.php
w****o-child/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/rating.php
w****o-child/woocommerce/single-product/related.php
w****o-child/woocommerce/single-product/review.php
w****o-child/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
w****o-child/woocommerce/single-product/short-description.php
w****o-child/woocommerce/single-product/stock.php
w****o-child/woocommerce/single-product/tabs/description.php
w****o-child/woocommerce/single-product/tabs/tabs.php
w****o-child/woocommerce/single-product/title.php
w****o-child/woocommerce/single-product/up-sells.php
w****o-child/woocommerce/single-product-reviews.php version 3.6.0 is out of date. The core version is 4.3.0
w****o-child/woocommerce/single-product.php

Outdated Templates: ❌
					
					
						Learn how to update

### Authorize.Net Credit Card ###

Environment: Production
Tokenization Enabled: ✔
Debug Mode: Save to Log
Webhooks Enabled: ❌
						No signature key configured

### Action Scheduler ###

Complete: 35,368
Oldest: 2020-08-14 17:43:00 -0700
Newest: 2020-09-14 17:37:20 -0700
`
```
</details>

### PR DESCRIPTION
**Pull request by vedanshujain**
We hold coupons when payment is failed if the setting "hold stock for checkout" is enabled for some minutes. This is to allow the customers to try again if they want, and to give time to complete payment for gateways where it could take some time.

Unfortunately this cause for some bad UX user retries by starting the order from scratch again, and then if they apply the coupon, usage limit gets hit because the earlier coupon is still held and is counted towards the usage. This PR improves the error message in these cases when usage is applied per customer and is hit, by stating to go to my account to complete/cancel payment (in case of logged in user) or to wait for some time in case of guest users.

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #27673.

### How to test the changes in this Pull Request:

(While logged in to an account)
1. Create a coupon with a usage limit of 1 (either per user, or a global usage limit).
2. Checkout using this coupon, but use a payment method which will payment step so that order is set to pending or failed. (For example, stripe)
3. Checkout using this coupon again, you will see a better error message which will ask you to go to my account and cancel or complete the order. If my account page is not enabled on the site, it will just say usage limit reached.

(As a guest)
1. Create a coupon with a usage limit of 1 (either per user, or a global usage limit).
2. Checkout using this coupon, but use a payment method which will payment step so that order is set to pending or failed. (For example, stripe)
3. Checkout using this coupon again, you will see a better error message which will ask you to wait for some time or contact support.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Fix - Better error messages when usage limit are reached.

### CODE REVIEW

File: includes/class-wc-coupon.php [hunk 1]
```diff
@@ -994,6 +996,17 @@ public function get_coupon_error( $err_code ) {
 			case self::E_WC_COUPON_NOT_APPLICABLE:
 				$err = __( 'Sorry, this coupon is not applicable to your cart contents.', 'woocommerce' );
 				break;
+			case self::E_WC_COUPON_USAGE_LIMIT_COUPON_STUCK:
+				if ( is_user_logged_in() && wc_get_page_id( 'myaccount' ) > 0 ) {
+					// translators: %s: myaccount page link.
```
Comment (roykho) [inline includes/class-wc-coupon.php:1001]: For translator comments, I think we generally use `/* */` comment format.

**Review by roykho — CHANGES_REQUESTED** [review]
Left a comment on coding standards. Besides that I think in general it tested well however if you try to checkout again with valid payment after 1st fail, you will get this message 

This message could be misleading as the coupon is currently just locked to their pending order. Perhaps we should add the same message to go to their my account to cancel the order to reinstate this coupon? This way it is more consistent?

The other thing I noticed was that the Travis CI is failing on the e2e tests but I don't think it is related to this PR.

File: includes/class-wc-coupon.php (hunk 1 above)
Comment (vedanshujain) [reply includes/class-wc-coupon.php]: thanks, fixed in https://github.com/woocommerce/woocommerce/pull/28592/commits/f248276a89d104db3258bc139f663393bbace98b

**Comment by vedanshujain** [comment]
>if you try to checkout again with valid payment after 1st fail, you will get this message 

@roykho Looks like this is coming via https://github.com/woocommerce/woocommerce/blob/master/includes/class-wc-cart.php#L726, which is basically eating up all the coupon error messages. 

I'd rather target this in a different PR targeted to next release instead of this one since it will affect all error messages instead of just this one case. And so will need more robust testing and tests as well. Do you think if its fine to target refactoring that method in a different PR?

**Comment by roykho** [comment]
> I'd rather target this in a different PR targeted to next release instead of this one since it will affect all error messages instead of just this one case.

Ok that's fine.

**Review by roykho — APPROVED** [review]
Approving this and a separate issue will be opened for the error message by author.

**Comment by vedanshujain** [comment]
Closing the loop: https://github.com/woocommerce/woocommerce/issues/28641

**Pull request merged**